src/6model/reprs/CStr.c

```diff
@@ -71,6 +71,17 @@ static void deserialize_stable_size(MVMThreadContext *tc, MVMSTable *st, MVMSeri
     st->size = sizeof(MVMCStr);
 }
 
+static void deserialize(MVMThreadContext *tc, MVMSTable *st, MVMObject *root, void *data, MVMSerializationReader *reader) {
+    MVMString *orig = MVM_serialization_read_str(tc, reader);
+    MVMCStrBody *body = (MVMCStrBody *)data;
+    MVM_ASSIGN_REF(tc, &(root->header), body->orig, orig);
```

Comment:
Isn't using MVM_ASSIGN_REF here actually overly cautious? The deserialized string will be allocated in gen2 directly, so a write barrier should never been needed.
Not saying that we should make it less cautious. Just checking my understanding.

Reply:
Your understanding is correct.